api/resource.yaml

```diff
@@ -0,0 +1,2200 @@
+                      - id: "7c4d5e6f-8f3a-4e93-c565-5e3ddf98hi6f"
+                        name: "Phone Booking"
+                        description: "Phone-based booking"
+                        handler: "phone-booking"
+                        identifier: null
+                        parent: "7a4b1f8e-5c69-4b60-9232-2b0aaf65ef3c"
+                        resourceServerId: "3fa85f64-5717-4562-b3fc-2c963f66afa6"
+                    links: []
+        "400":
+          description: Bad request
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+              examples:
+                invalid-limit:
+                  summary: Invalid limit parameter
+                  value:
+                    code: "RES-1011"
+                    message: "Invalid limit parameter"
+                    description: "The limit parameter must be a positive integer"
+                invalid-offset:
+                  summary: Invalid offset parameter
+                  value:
+                    code: "RES-1012"
+                    message: "Invalid offset parameter"
+                    description: "The offset parameter must be a non-negative integer"
+        "404":
+          description: Resource server not found
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+              example:
+                code: "RES-1003"
+                message: "Resource server not found"
+                description: "The resource server with the specified id does not exist"
+        "500":
+          description: Internal server error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+              example:
+                code: "RES-5000"
+                message: "Internal server error"
+                description: "An unexpected error occurred while processing the request"
+
+    post:
+      tags:
+        - resources
+      summary: Create a new resource
+      description: |
+        Creates a new resource within the resource server. 
+        - Set parent to null for top-level resources
+        - Set parent to UUID for nested resources
+      parameters:
+        - in: path
+          name: rsId
+          required: true
+          schema:
+            type: string
+            format: uuid
+          description: Resource server ID
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/CreateResourceRequest'
+            examples:
+              top-level-resource:
+                summary: Top-level resource
+                value:
+                  name: "Reservations"
+                  description: "Reservation management"
+                  handler: "reservations"
```

Comment:
Should this be `handler` or `handle`? We're using `handle` in the ou APIs. Better to stick to one.

@darshanasbg WDYT?

Reply:
settled to use `handle`